tools/DocsLinter/Program.cs

```diff
@@ -0,0 +1,252 @@
+                    {
+                        LogInvalidLink(markdownFilePath, localLink);
+                        return false;
+                    }
+
+                    // Check that the heading is valid.
+                    if (!otherMarkdownDoc.Headings.Contains(localLink.Heading.Value))
+                    {
+                        LogInvalidLink(markdownFilePath, localLink);
+                        return false;
+                    }
+                }
+                else
+                {
+                    // May be a markdown file (without a heading) or non-markdown file. E.g. - "../README.md" or "../../spatialos.json".
+                    if (!markdownFiles.ContainsKey(fullyQualifiedFilePath) &&
+                        !File.Exists(fullyQualifiedFilePath))
+                    {
+                        LogInvalidLink(markdownFilePath, localLink);
+                        return false;
+                    }
+                }
+            }
+
+            return true;
+        }
+        /// <summary>
+        /// A helper function that checks the validity of a single remote link.
+        /// Side effects: Prints to the console.
+        /// </summary>
+        /// <param name="markdownFilePath">The fully qualified path of the Markdown file to check</param>
+        /// <param name="remoteLink">The object representing the remote link to check.</param>
+        /// <returns>A bool indicating success/failure</returns>
+        internal static bool CheckRemoteLink(string markdownFilePath, RemoteLink remoteLink)
+        {
+            try
+            {
+                var request = WebRequest.CreateHttp(remoteLink.Url);
+                request.Method = "GET";
+                var response = request.GetResponse() as HttpWebResponse;
+                response.Close();
+
+                // Check for non-200 error codes.
+                if (response.StatusCode != HttpStatusCode.OK)
+                {
+                    LogLinkWarning(markdownFilePath, remoteLink, $"returned a status code of: {(int)response.StatusCode}");
+                }
+
+                return true;
+            }
+            catch (WebException ex)
+            {
+                // There was an error code. Check if it was a 404.
+                // Any other 4xx errors are considered "okay" for now.
+                if (ex.Status == WebExceptionStatus.ProtocolError)
+                {
+                    var statusCode = ((HttpWebResponse) ex.Response).StatusCode;
+                    if (statusCode == HttpStatusCode.NotFound)
+                    {
+                        LogInvalidLink(markdownFilePath, remoteLink);
+                        return false;
+                    }
+
+                    LogLinkWarning(markdownFilePath, remoteLink, $"returned a status code of: {(int)statusCode}");
+                    return true;
+                }
+
+                Console.Error.WriteLine(ex.ToString());
```

Comment:
Collapse into a function here and other places?

Reply:
Yes probably a good idea